src/pages/CollectionSingleViewPage.tsx

```diff
@@ -359,13 +402,149 @@ const TabPanel = ({ children, value, index, ...other }: TabPanelProps) => {
   );
 };
 
+type MutationTableTabContentProps = {
+  mutationType: string;
+  locationSelector: LocationSelector;
+  variants: { query: VariantSelector; name: string; description: string }[];
+  dateRangeSelector: DateRangeSelector;
+  mode: 'Single' | 'CompareToBaseline';
+  baselineVariant?: VariantSelector;
+};
+
+const MutationTableTabContent = ({
+  mutationType,
+  variants,
+  locationSelector,
+  dateRangeSelector,
+  mode,
+  baselineVariant,
+}: MutationTableTabContentProps) => {
+  const mutations = useQuery(
+    async signal => {
+      if (!variants) {
+        return undefined;
+      }
+      const data: MutationObject[] = [];
+
+      await Promise.allSettled(
+        variants.map(variant => {
+          const selector: LapisSelector = addDefaultHostAndQc({
+            location: locationSelector,
+            variant: variant.query,
+            samplingStrategy: SamplingStrategy.AllSamples,
+            dateRange: dateRangeSelector,
+          });
+
+          return Promise.all([
+            fetchSamplesCount(selector, signal),
+            MutationProportionData.fromApi(selector, 'aa', signal),
+            MutationProportionData.fromApi(selector, 'nuc', signal),
+          ]).then(async ([variantCount, aaMutationDataset, nucMutationDataset]) => {
+            data.push({
+              ...variant,
+              variantCout: variantCount,
+              aaMutations: aaMutationDataset.payload,
+              nucMutations: nucMutationDataset.payload,
+            });
+          });
+        })
+      );
+
+      return data;
+    },
+    [variants, locationSelector, dateRangeSelector]
+  );
+
+  const baselineMutations = useQuery(
+    async signal => {
```

Comment:
The problem, why sometimes data doesn't load, is: This async function isn't actually async, since it does not `await` anything. 

How to reproduce it consistently:
* choose a collection
* select a baseline
* switch to the "mutations" tab

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

"Top-level code, up to and including the first await expression (if there is one), is run synchronously. In this way, an async function without an await expression will run synchronously."
-> The function as it is now will initialize an empty data object, construct a promise and immediately return the empty data. 

I'd propose to restructure this function to use the promises in a more idiomatic way:
```
  const baselineMutations = useQuery(
    signal => {
      if (mode !== 'CompareToBaseline') {
        return Promise.resolve({
          aaMutations: [],
          nucMutations: [],
        });
      }

      const selector: LapisSelector = addDefaultHostAndQc({
        location: locationSelector,
        variant: baselineVariant,
        samplingStrategy: SamplingStrategy.AllSamples,
        dateRange: dateRangeSelector,
      });

      return Promise.all([
        MutationProportionData.fromApi(selector, 'aa', signal),
        MutationProportionData.fromApi(selector, 'nuc', signal),
      ]).then(async ([aaMutationDataset, nucMutationDataset]) => {
        return {
          aaMutations: aaMutationDataset.payload,
          nucMutations: nucMutationDataset.payload,
        };
      });
    },
    [baselineVariant, mode, locationSelector, dateRangeSelector]
  );
```

The same applies to the function in the "mutations useQuery". The usage of promises mirrors more closely the flow of the data instead of setting data via side effects. (Also notice that the function doesn't need to be explicitly `async` because it returns a Promise anyway.)

The problem here can also be fixed by simply `await`ing the Promise in the async function.